### PR TITLE
update readme and message formatting

### PR DESCRIPTION
**Pull request by catherinewu**

### CODE REVIEW

File: tau_bench/envs/user.py
Comment (catherinewu) [inline tau_bench/envs/user.py:1]: I updated to `self.messages.append(message.model_dump())`. Before, I was getting errors (with Anthropic models) that there were back-to-back user messages. This fixes the errors.

File: tau_bench/model_utils/api/sample.py [hunk 1]
```diff
@@ -94,15 +93,15 @@ def execute(self, invocable_or_invokables: Callable[..., T] | list[Callable[...,
         if self.panic_on_first_model_error:
             if isinstance(invocable_or_invokables, Callable):
                 results = list(
-                    func_tools.map(
+                    functools.map(
```
Comment (catherinewu) [inline tau_bench/model_utils/api/sample.py:96]: I didn't see func_tools defined so I think this is just fixing a typo?

Comment (noahshinn) [reply tau_bench/model_utils/api/sample.py:96]: Oh I was missing a module. I will add it in a follow-up commit

**Review by noahshinn — APPROVED** [review]
Thanks for the fixes!

**Pull request merged**